player/lua/ytdl_hook.lua

```diff
@@ -740,6 +804,13 @@ local function add_single_video(json)
             "http_proxy", json.proxy)
     end
 
+    if cookies and cookies ~= "" then
+        local parsed_cookies = parse_cookies(cookies)
+        local lavf_cookies = serialize_cookies_for_avformat(parsed_cookies)
+        stream_opts = append_libav_opt(stream_opts,
+            "cookies", lavf_cookies)
+    end
+
```

Comment:
Pretty sure this entire part is superfluous and unneeded.

Reply:
How? I don't see any `set_cookies()` invocations that would happen in the same code paths where `add_single_video()` is invoked.

Reply:
Sorry you're right, I read it wrong. Although you should just be able to use `set_cookies` in there I think.

Reply:
I think it's better to be consistent, but I can change it if you want.

Reply:
It's the same as how https headers are handled.

Reply:
HTTP headers access a different property. I didn't want to read and write to a single property twice. Furthermore, I just changed this code to override existing cookies (as opposed to `set_cookies()` which preserves existing cookies). 🤷